Classes/Domain/Repository/DocumentRepository.php

```diff
@@ -645,4 +645,168 @@ private function findSolr($collections, $settings, $searchParams, $listedMetadat
         $search->prepare();
         return $search;
     }
+
+    /**
+     * Find the uid of the previous document relative to the current document uid.
+     * Otherwise backtrack the closest previous leaf node.
+     *
+     * @access public
+     *
+     * @param int $uid
+     *
+     * @return int|null
+     */
+    public function getPreviousDocumentUid($uid)
+    {
+        $currentDocument = $this->findOneByUid($uid);
+        $currentVolume = '';
+        $parentId = $currentDocument->getPartof();
```

Comment:
This change will only work in case when documents are stored in database, but in case of DFG Viewer they are passed as XML files. It causes error:
`Call to a member function getPartof() on null`

It is necessary to make a null check already on `$currentDocument` variable.

Reply:
@michaelkubina Please apply your fix from #1381 for this pull request as well.